internal/datastore/postgres/stats.go

```diff
@@ -0,0 +1,75 @@
+package postgres
+
+import (
+	"context"
+	"fmt"
+
+	sq "github.com/Masterminds/squirrel"
+
+	"github.com/authzed/spicedb/internal/datastore"
+)
+
+const (
+	tableMetadata = "metadata"
+	colUniqueID   = "unique_id"
+
+	tablePGClass = "pg_class"
+	colReltuples = "reltuples"
+	colRelname   = "relname"
+)
+
+var (
+	queryUniqueID          = psql.Select(colUniqueID).From(tableMetadata)
+	queryEstimatedRowCount = psql.
+				Select(colReltuples).
+				From(tablePGClass).
+				Where(sq.Eq{colRelname: tableTuple})
+)
+
+func (pgd *pgDatastore) Statistics(ctx context.Context) (datastore.Stats, error) {
+	tx, err := pgd.dbpool.Begin(ctx)
+	if err != nil {
+		return datastore.Stats{}, fmt.Errorf("unable to establish transaction: %w", err)
+	}
+	defer tx.Rollback(ctx)
+
+	if pgd.analyzeBeforeStatistics {
+		if _, err := tx.Exec(ctx, fmt.Sprintf("ANALYZE %s", tableTuple)); err != nil {
```

Comment:
Is this a query that can use the $1 syntax?

Reply:
It doesn't work, I think because this is technically DDL.